validity/src/proposer.rs

```diff
@@ -295,15 +320,72 @@ where
             let (status, proof) =
                 self.driver_config.network_prover.get_proof_status(proof_request_id).await?;
 
+            let request_details =
+                self.driver_config.network_prover.get_proof_request(proof_request_id).await?;
+
             // Check if current time exceeds deadline. If so, the proof has timed out.
             let current_time = std::time::SystemTime::now()
                 .duration_since(std::time::UNIX_EPOCH)
                 .unwrap()
                 .as_secs();
-            if current_time > status.deadline {
+
+            // Cancel the request in the network if the auction timeout is exceeded.
+            if let Some(request_details) = request_details {
+                let auction_deadline =
+                    request_details.created_at + self.requester_config.auction_timeout;
+                if request_details.fulfillment_status == FulfillmentStatus::Requested as i32 &&
+                    current_time < status.deadline() &&
+                    current_time > auction_deadline
+                {
+                    // Cancel the request in the network.
+                    self.driver_config.network_prover.cancel_request(proof_request_id).await?;
+
+                    // Mark the request as cancelled in the database.
+                    match self
+                        .proof_requester
+                        .handle_failed_request(request.clone(), status.execution_status(), true)
```

Comment:
Can we just add handle_cancelled_request function and revert the change made to add boolean flag?